### PR TITLE
Upgrade dependencies to fix high vulnerabilities

### PR DESCRIPTION
**Pull request by samanthagottlieb**
## What does this change?

This upgrades dependencies to resolve [Snyk](https://app.snyk.io/org/guardian/project/8ecf100e-b5f7-43e5-bbaa-c28fcd28eb7a) high vulnerabilities. Some dependencies have been manually overriden as the libraries dependent on them were using versions with vulnerabilities.

The `sbt-dependency-graph` plugin has been added to ensure the Snyk CLI and dashboard are accurate. See more [here](https://support.snyk.io/hc/en-us/articles/360004167317-How-to-install-the-SBT-dependency-graph-plugin-to-test-Scala-projects-with-Snyk-CLI).

**Note**: there is one outstanding high vulnerability (`io.netty:netty-codec-http2`) introduced through the Guardian's `simple-configuration-ssm` package. There is an open [PR](https://github.com/guardian/simple-configuration/pull/63) addressing this. Once this is merged, the version of `simple-configuration-ssm` can be upgraded to resolve this.

## How to test

- Run the rule manager and checker apps locally, following README instructions. Do they run as expected?
- Deploy this branch to CODE and check that [rule manager](https://manager.typerighter.code.dev-gutools.co.uk/) and [checker](https://checker.typerighter.code.dev-gutools.co.uk/) run as expected

## How can we measure success?

There are no longer any high vulnerabilities (with the exception of the `io.netty:netty-codec-http2` vulnerability) in the Snyk [typerighter](https://app.snyk.io/org/guardian/project/8ecf100e-b5f7-43e5-bbaa-c28fcd28eb7a) project.

### CODE REVIEW

**Review by jonathonherbert — COMMENTED** [review]
This is great! One comment about dependencyTree, and it'll need a check in CODE to make sure everything's working as expected. Back in on Monday and happy to do that as part of the review.

One thing we'll need to check when we're upgrading LanguageTool – whether it affects any of the built-in rules we depend upon. 

There's a bit in the README [here](https://github.com/guardian/typerighter#upgrading-languagetool) that I think we could expand on!

If it'd be useful, be really happy to pair on this – this is a bit obscure, and I want to make sure the docs and process are as sane as possible 🙏

File: project/plugins.sbt [hunk 1]
```diff
@@ -5,5 +5,6 @@ addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.9.0")
 addSbtPlugin("com.typesafe.sbt" % "sbt-less" % "1.1.2")
 addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.0") // "2.4.0" is just sbt plugin version
 addSbtPlugin("org.scalikejdbc" %% "scalikejdbc-mapper-generator" % "3.5.0")
+addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.10.0-RC1")
```
[the remark this answers is not in the document]
Comment (samanthagottlieb) [reply project/plugins.sbt:8]: Yes @rebecca-thompson and I discovered this, however without the plugin explicitly added, Snyk was reporting a vulnerability that wasn't reflected in the dependency tree. We tried upgrading sbt and the Snyk CLI, but it was still needed to make sure vulnerabilities were reported accurately.

**Comment by samanthagottlieb** [comment]
> This is great! One comment about dependencyTree, and it'll need a check in CODE to make sure everything's working as expected. Back in on Monday and happy to do that as part of the review.
> 
> One thing we'll need to check when we're upgrading LanguageTool – whether it affects any of the built-in rules we depend upon.
> 
> There's a bit in the README [here](https://github.com/guardian/typerighter#upgrading-languagetool) that I think we could expand on!
> 
> If it'd be useful, be really happy to pair on this – this is a bit obscure, and I want to make sure the docs and process are as sane as possible 🙏

I've deployed to CODE and checked that the rule manager and checker are running as usual 👍.

Thanks for the link to the Language Tool section of README. Pairing on this on Monday would be great!

File: project/plugins.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathonherbert) [reply project/plugins.sbt]: Oh wow! Thanks for spotting this. I can see snyk reporting 

```
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONDATAFORMAT-1047329] in com.fasterxml.jackson.dataformat:jackson-dataformat-cbor@2..
```

... initially I thought the `..` at the end was a misreport, but I had a dig and it's the [output from the graph being truncated](https://stackoverflow.com/questions/37412934/how-to-make-sbt-not-truncate-its-output) 😅

More digging and it turns out that [Snyk knows about this](https://support.snyk.io/hc/en-us/articles/9590215676189-Deeply-nested-Scala-projects-have-dependencies-truncated), I've added what I hope to be the solution, verified the output w/ `snyk test` and removed the plugin in [51ba3cd](https://github.com/guardian/typerighter/pull/458/commits/51ba3cd1bcbfb240634145d6ecc05068d7f96a27) – does that look right to you?

[the remark this answers is not in the document]
Comment (samanthagottlieb) [reply project/plugins.sbt]: Ah thank you so much for adding this - yes that looks like it's fixed it! 👏

**Comment by samanthagottlieb** [comment]
Just a note that I've deployed this to CODE, tested the rule manager and checker, and also tested running TR in Composer CODE. All seems to be working as expected!

**Review by jonathonherbert — APPROVED** [review]

**Pull request merged**